src/IO/Resources/AnimationsLoader.cs

```diff
@@ -982,42 +904,27 @@ public bool PixelCheck(ushort animID, byte group, byte direction, bool uop, int
 
         private static uint _lastFlags = 0xFFFFFFFF;
 
+        public bool FlagsApplied => _lastFlags != 0xFFFFFFFF;
+
         public void UpdateAnimationTable(uint flags)
         {
-            if (_lastFlags != 0xFFFFFFFF && flags != _lastFlags)
+            if (flags != _lastFlags)
             {
-                /* This happens when you log out of an account then into another
-                 * one with different expansions activated. Just reload the anim
-                 * files from scratch. */
-                Array.Clear(DataIndex, 0, DataIndex.Length);
-                LoadInternal();
-            }
-
-            _lastFlags = flags;
-
-            for (ushort i = 0; i < Constants.MAX_ANIMATIONS_DATA_INDEX_COUNT; i++)
-            {
-                bool replace = DataIndex[i].FileIndex >= 3;
-
-                if (DataIndex[i].FileIndex == 1)
+                if (_lastFlags != 0xFFFFFFFF)
                 {
-                    // TODO: Find out what client version introduced this negative feature flag change.
-                    //replace = (World.ClientLockedFeatures.Flags & (LockedFeatureFlags.TheSecondAge | LockedFeatureFlags.Renaissance)) == 0;
-                    replace = (World.ClientLockedFeatures.Flags & LockedFeatureFlags.LordBlackthornsRevenge) != 0;
-                }
-                else if (DataIndex[i].FileIndex == 2)
-                {
-                    replace = (World.ClientLockedFeatures.Flags & LockedFeatureFlags.AgeOfShadows) != 0;
+                    /* This happens when you log out of an account then into another
+                     * one with different expansions activated. Just reload the anim
+                     * files from scratch. */
+                    Array.Clear(DataIndex, 0, DataIndex.Length);
+                    LoadInternal();
                 }
 
-                if (replace)
-                {
-                    if (DataIndex[i].BodyConvGroups != null)
-                    {
-                        DataIndex[i].HasBodyConversion = true;
-                    }
-                }
+                ProcessBodyConvDef((LockedFeatureFlags)flags);
+                ProcessBodyDef();
+                ProcessCorpseDef();
```

Comment:
Not sure if it's volutary, but from what we said on discord `Body.def` & `Corpse.def` should be loaded before `Bodyconv.def` [?]

Reply:
I thought so initially but that's not the case. You ran a test with the following body.def:

```
11 {28} 1401
```

And bodyconv.def
```
28    206  -1    -1    -1    -1
```

Then, drawing body 11 draws a spider (anim.mul+28) not a swamp dragon (anim2.mul+206). That must mean the body.def overwrites the bodyconv.def, so it is loaded second.

Reply:
I think the main thing we were all missing is that there's a "logical" set of body values and then physical offsets into files. Body.def first column is logical body value and the number inside {} is the physical offset into anim.mul, not a logical body value. So there is never any sort of "chaining" going on in these files and hence why my patch removes the loops.

Reply:
Which body is used also depends on feature flags (for the EA client at least).